src/core_modules/capture-core/components/Pages/MainPage/EventWorkingLists/types/eventWorkingLists.types.js

```diff
@@ -42,3 +42,25 @@ export type EventWorkingListsTemplate = {
 };
 
 export type EventWorkingListsTemplates = Array<EventWorkingListsTemplate>;
+
+export type ColumnConfigBase = {
+    id: string,
+    visible: boolean,
+    type: $Values<typeof dataElementTypes>,
+    header: string,
+    options?: ?Array<{text: string, value: any}>,
+    multiValueFilter?: boolean,
+};
+export type MetadataColumnConfig = {
+    ...ColumnConfigBase,
+};
+
+export type MainColumnConfig = {
+    ...ColumnConfigBase,
+    isMainProperty: true,
+    apiName?: string,
+};
```

Comment:
Wanna see if it makes sense to put some effort into annotating this strictly?

Reply:
To be more precise: This causes a problem since the columnConfigs are not restructured before being passed on to the `WorkingLists`.